### DISCUSSION TITLE
Mission equality check failed

### DISCUSSION THREAD

**Discussion by Kokhirus** [FakeFishGames/Barotrauma#14083] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

While connecting one random server, I got the same exception.
When connected to lobby, got this exception:
![изображение_2024-06-03_145740037](https://github.com/FakeFishGames/Barotrauma/assets/91697233/d73b6b2d-7c4f-4c39-822e-8380e5cc68d4)
After clicking "Join" button, got both of these exceptions consecutively:
![image](https://github.com/FakeFishGames/Barotrauma/assets/91697233/af01863d-01e0-4783-b929-9f908df99714)
![image](https://github.com/FakeFishGames/Barotrauma/assets/91697233/236e647d-36d2-481c-add8-50a82ccca1e7)

### Reproduction steps

1. Join the server lobby
2. Join the session

### Bug prevalence

Just once

### Single player or multiplayer?

Multiplayer hosted using a dedicated server

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Astro1247** [comment]
Same here. Using v1.5.9.1
Dedicated server
No mods

![image](https://github.com/user-attachments/assets/9711c0af-6340-4705-9ddf-5ea97d51b49f)
